initrd/bin/flash.sh

```diff
@@ -13,194 +13,253 @@ echo
+    whiptail --title "Confirmation" --yesno "You are about to flash the following rom:\n\n$1\n\nSHA256SUM: $ROM_HASH\n\nDo you want to continue?" 10 60
+    if [ $? -ne 0 ]; then
+      die "User cancelled flashing"
     fi
+  else
+    # Console prompt if whiptail is not available
+    echo "You are about to flash the following rom:"
+    echo "$ROM"
+    echo "SHA256SUM: $ROM_HASH"
+    echo ""
+    echo "Do you want to proceed flashing this rom? (y/N)"
+    read -r -n 1 -s -p "" key
+    if [ "$key" != "y" ]; then
+      die "User cancelled flashing"
+    fi
+  fi
 }
 
 flash_rom() {
   ROM=$1
   if [ "$READ" -eq 1 ]; then
-    flashrom $CONFIG_FLASHROM_OPTIONS -r "${ROM}" \
-    || die "Backup to $ROM failed"
+    flashrom $CONFIG_FLASHROM_OPTIONS -r "${ROM}" ||
+      die "Backup to $ROM failed"
   else
+    # If we have a rom file, we ask confirmation to the user if flash-gui.sh didn't provided -i 1 as flash.sh argument
+    if [ "$INTEGRITY_VERIFIED" -eq 0 ]; then
+      confirm_rom_sha256sum_hash "$ROM"
+    fi
+
     cp "$ROM" /tmp/${CONFIG_BOARD}.rom
-    sha256sum /tmp/${CONFIG_BOARD}.rom
     if [ "$CLEAN" -eq 0 ]; then
-      preserve_rom /tmp/${CONFIG_BOARD}.rom \
-      || die "$ROM: Config preservation failed"
+      preserve_rom /tmp/${CONFIG_BOARD}.rom ||
+        die "$ROM: Config preservation failed"
     fi
     # persist serial number from CBFS
-    if cbfs.sh -r serial_number > /tmp/serial 2>/dev/null; then
+    if cbfs.sh -r serial_number >/tmp/serial 2>/dev/null; then
       echo "Persisting system serial"
       cbfs.sh -o /tmp/${CONFIG_BOARD}.rom -d serial_number 2>/dev/null || true
       cbfs.sh -o /tmp/${CONFIG_BOARD}.rom -a serial_number -f /tmp/serial
     fi
     # persist PCHSTRP9 from flash descriptor
     if [ "$CONFIG_BOARD" = "librem_l1um" ]; then
       echo "Persisting PCHSTRP9"
-      flashrom $CONFIG_FLASHROM_OPTIONS -r /tmp/ifd.bin --ifd -i fd >/dev/null 2>&1 \
-      || die "Failed to read flash descriptor"
+      flashrom $CONFIG_FLASHROM_OPTIONS -r /tmp/ifd.bin --ifd -i fd >/dev/null 2>&1 ||
+        die "Failed to read flash descriptor"
       dd if=/tmp/ifd.bin bs=1 count=4 skip=292 of=/tmp/pchstrp9.bin >/dev/null 2>&1
       dd if=/tmp/pchstrp9.bin bs=1 count=4 seek=292 of=/tmp/${CONFIG_BOARD}.rom conv=notrunc >/dev/null 2>&1
     fi
 
     flashrom $CONFIG_FLASHROM_OPTIONS -w /tmp/${CONFIG_BOARD}.rom \
-      -V -o "/tmp/flashrom-$(date '+%Y%m%d-%H%M%S').log" 2>&1 | \
-      flashrom_progress "$(stat -c %s "/tmp/${CONFIG_BOARD}.rom")" \
-      || die "$ROM: Flash failed"
+      -V -o "/tmp/flashrom-$(date '+%Y%m%d-%H%M%S').log" 2>&1 |
+      flashrom_progress "$(stat -c %s "/tmp/${CONFIG_BOARD}.rom")" ||
+      die "$ROM: Flash failed"
   fi
 }
 
-if [ "$1" == "-c" ]; then
-  CLEAN=1
-  READ=0
-  ROM="$2"
-elif [ "$1" == "-r" ]; then
-  CLEAN=0
-  READ=1
-  ROM="$2"
-  touch $ROM
-else
-  CLEAN=0
-  READ=0
-  ROM="$1"
+#Refactor argument parsing so arguments are not positional
```

Comment:
Nitpick, the comments here shouldn't be worded with "refactor" since this refactoring has been done at this point